homeassistant/components/lifx/coordinator.py

```diff
@@ -115,6 +115,52 @@ def current_infrared_brightness(self) -> str | None:
         """Return the current infrared brightness as a string."""
         return infrared_brightness_value_to_option(self.device.infrared_brightness)
 
+    async def diagnostics(self) -> dict[str, Any]:
+        """Return diagnostic information about the device."""
+        features = lifx_features(self.device)
+        device_data = {
+            "firmware": self.device.host_firmware_version,
+            "vendor": self.device.vendor,
+            "product_id": self.device.product,
+            "features": features,
+            "hue": self.device.color[0],
+            "saturation": self.device.color[1],
+            "brightness": self.device.color[2],
+            "kelvin": self.device.color[3],
+            "power": self.device.power_level,
+        }
+
+        if features["multizone"] is True:
+
+            if features["extended_multizone"] is True:
+                await self.async_get_extended_color_zones()
+            else:
+                await self.async_get_color_zones()
+
```

Comment:
```suggestion
```

Generally we don't want diagnostics to update the device again as its changing the state of the data which makes it difficult to capture a problem if the user is trying to get diagnostics because something is broken at a specific time.